frame/oracle/src/tests.rs

```diff
@@ -20,205 +20,462 @@ use sp_runtime::{
+                        asset_info.block_interval,
+                        asset_info.reward,
+                        asset_info.slash,
+                    ),
+                    BadOrigin
+                );
+                Ok(())
+            })?;
+        }
+
+
+        #[test]
+        fn can_have_multiple_assets(
+            asset_id_1 in asset_id(),
+            asset_id_2 in asset_id(),
+            asset_info_1 in asset_info(),
+            asset_info_2 in asset_info(),
+        ) {
+            new_test_ext().execute_with(|| {
+                let root_account = get_root_account();
+
+                prop_assert_ok!(Oracle::add_asset_and_info(
+                    Origin::signed(root_account),
+                    asset_id_1,
+                    asset_info_1.threshold,
+                    asset_info_1.min_answers,
+                    asset_info_1.max_answers,
+                    asset_info_1.block_interval,
+                    asset_info_1.reward,
+                    asset_info_1.slash,
+                ));
+
+                prop_assert_ok!(Oracle::add_asset_and_info(
+                    Origin::signed(root_account),
+                    asset_id_2,
+                    asset_info_2.threshold,
+                    asset_info_2.min_answers,
+                    asset_info_2.max_answers,
+                    asset_info_2.block_interval,
+                    asset_info_2.reward,
+                    asset_info_2.slash,
+                ));
+
+                prop_assert_eq!(Oracle::asset_info(asset_id_1), Some(asset_info_1));
+                prop_assert_eq!(Oracle::asset_info(asset_id_2), Some(asset_info_2));
+                prop_assert_eq!(Oracle::assets_count(), 2);
+
+                Ok(())
+            })?;
+        }
+
+
+
+        #[test]
+        fn max_answers_cannot_be_less_than_min_answers(
+            asset_id in asset_id(),
+            asset_info in asset_info(),
+        ) {
+            let root_account = get_root_account();
+
+            new_test_ext().execute_with(|| {
+                prop_assert_noop!(
+                    Oracle::add_asset_and_info(
+                        Origin::signed(root_account),
+                        asset_id,
+                        asset_info.threshold,       // notice that max and min are reversed:
+                        asset_info.max_answers,     // MIN
+                        asset_info.min_answers - 1, // MAX
+                        asset_info.block_interval,
+                        asset_info.reward,
+                        asset_info.slash,
+                    ),
+                    Error::<Test>::MaxAnswersLessThanMinAnswers
+                );
+
+                Ok(())
+            })?;
+        }
+
+
+
+        #[test]
+        fn threshold_cannot_be_100(
+            asset_id in asset_id(),
+            asset_info in asset_info(),
+        ) {
+            let root_account = get_root_account();
+
+            new_test_ext().execute_with(|| {
```

Comment:
probably not prop test too. OR proptest but should be on different variable - on precentage :)

Reply:
I think that makes sense. The 100% one was because it was migrated from an older test. Makes sense to make the percentage a prop